File: chap11/fig11-21.py
import pandas
import cmdstanpy
import numpy as np
from plotnine import ggplot, aes, theme_bw, facet_wrap, facet_grid, geom_line, geom_ribbon, labs

fit = cmdstanpy.from_csv('output/result-model11-13')
d_ms = fit.draws_pd().filter(regex='^(trend|calendar|weather|event|ar)\\[\d+\\]$')
qua = np.quantile(d_ms, [0.025, 0.25, 0.50, 0.75, 0.975], axis=0)
d_est = pandas.DataFrame(qua.T, columns=['2.5%', '25%', '50%', '75%', '97.5%'])
d_est['Parameter'] = d_ms.columns.str.extract('([a-z]+)')
d_est['Parameter'] = pandas.Categorical(d_est.Parameter, categories=['trend', 'calendar', 'weather', 'event', 'ar'])
d_est['X'] = d_ms.columns.str.extract('(\d+)').astype('int')

p = (ggplot()
    + theme_bw(base_size=18)
    + facet_grid('Parameter ~ .', scales='free_y')
    + geom_ribbon(d_est, aes('X', ymin='2.5%', ymax='97.5%'), alpha=1/6)
    + geom_ribbon(d_est, aes('X', ymin='25%',  ymax='75%'),   alpha=2/6)
    + geom_line(d_est, aes('X', '50%'))
    + labs(x='Time (Day)', y='')
)
p.save(filename='output/fig11-21.py.png', dpi=300, width=8, height=6)

File: chap11/fig11-21.R
library(dplyr)
library(ggplot2)

fit <- readRDS('output/result-model11-13.RDS')
d_ms <- fit$draws(format='df') %>% 
  select(starts_with(c('trend[', 'calendar[', 'weather[', 'event[', 'ar[')))
qua <- apply(d_ms, 2, quantile, probs=c(0.025, 0.25, 0.50, 0.75, 0.975))
d_est <- qua %>% t() %>% 
  data.frame(check.names=FALSE) %>% 
  tibble::rownames_to_column(var='Parameter_raw') %>% 
  tidyr::separate(Parameter_raw, into=c('Parameter', 'X'), sep='[\\[\\]]', convert=TRUE) %>% 
  mutate(Parameter=factor(Parameter, levels=c('trend', 'calendar', 'weather', 'event', 'ar')))
  
p <- ggplot() +
  theme_bw(base_size=18) +
  facet_grid(Parameter ~ ., scales='free_y') +
  geom_ribbon(data=d_est, aes(x=X, ymin=`2.5%`, ymax=`97.5%`), alpha=1/6) +
  geom_ribbon(data=d_est, aes(x=X, ymin=`25%`,  ymax=`75%`),   alpha=2/6) +
  geom_line(data=d_est, aes(x=X, y=`50%`)) +
  labs(x='Time (Day)', y='')
ggsave(p, file='output/fig11-21.png', dpi=300, w=8, h=6)
